ray-operator/controllers/raycluster_controller_fake_test.go

```diff
@@ -0,0 +1,425 @@
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			WorkerGroupSpecs: []rayiov1alpha1.WorkerGroupSpec{
+				{
+					Replicas:    pointer.Int32Ptr(expectReplicaNum),
+					MinReplicas: pointer.Int32Ptr(0),
+					MaxReplicas: pointer.Int32Ptr(10000),
+					GroupName:   groupNameStr,
+					RayStartParams: map[string]string{
+						"port":           "6379",
+						"redis-password": "LetMeInRay",
+						"num-cpus":       "1",
+					},
+					Template: corev1.PodTemplateSpec{
+						Spec: corev1.PodSpec{
+							Containers: []corev1.Container{
+								{
+									Name:    "ray-worker",
+									Image:   "rayproject/autoscaler",
+									Command: []string{"echo"},
+									Args:    []string{"Hello Ray"},
+									Env: []corev1.EnvVar{
+										{
+											Name: "MY_POD_IP",
+											ValueFrom: &corev1.EnvVarSource{
+												FieldRef: &corev1.ObjectFieldSelector{
+													FieldPath: "status.podIP",
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+					ScaleStrategy: rayiov1alpha1.ScaleStrategy{
+						WorkersToDelete: []string{
+							"pod1",
+							"pod2",
+						},
+					},
+				},
+			},
+		},
+	}
+
+	instanceReqValue := []string{instanceName}
+	instanceReq, err := labels.NewRequirement(
+		common.RayClusterLabelKey,
+		selection.Equals,
+		instanceReqValue)
+	assert.Nil(t, err, "Fail to create requirement")
+	groupNameReqValue := []string{groupNameStr}
+	groupNameReq, err := labels.NewRequirement(
+		common.RayNodeGroupLabelKey,
+		selection.Equals,
+		groupNameReqValue)
+	assert.Nil(t, err, "Fail to create requirement")
+
+	workerSelector = labels.NewSelector().Add(*instanceReq).Add(*groupNameReq)
+}
+
+func tearDown(t *testing.T) {
+	PrioritizeWorkersToDelete = false
+}
+
+func TestReconcile_RemoveWorkersToDelete_OK(t *testing.T) {
+	setupTest(t)
+	defer tearDown(t)
+
+	fakeClient := clientFake.NewClientBuilder().WithRuntimeObjects(testPods...).Build()
+
+	podList := corev1.PodList{}
+	err := fakeClient.List(context.Background(), &podList, client.InNamespace(namespaceStr))
+
+	assert.Nil(t, err, "Fail to get pod list")
+	assert.Equal(t, len(testPods), len(podList.Items), "Init pod list len is wrong")
+
+	testRayClusterReconciler := &RayClusterReconciler{
+		Client:   fakeClient,
+		Recorder: &record.FakeRecorder{},
+		Scheme:   scheme.Scheme,
+		Log:      ctrl.Log.WithName("controllers").WithName("RayCluster"),
+	}
+
+	err = testRayClusterReconciler.reconcilePods(testRayCluster)
+	assert.Nil(t, err, "Fail to reconcile Pods")
+
+	err = fakeClient.List(context.Background(), &podList, &client.ListOptions{
+		LabelSelector: workerSelector,
+		Namespace:     namespaceStr,
+	})
+
+	assert.Nil(t, err, "Fail to get pod list after reconcile")
+
+	assert.Equal(t, int(expectReplicaNum), len(podList.Items),
+		"Replica number is wrong after reconcile expect %d actual %d", expectReplicaNum, len(podList.Items))
+}
```

Comment:
Should we also check that WorkersToDelete has been cleared at the end?

Reply:
Good point. Will add it.

Reply:
As I mentioned earlier, we should make sure WorkersToDelete has been deleted in the CR - not just in testRayCluster